server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -738,23 +439,31 @@ private void ensureStorageGroup(PartialPath path) throws MetadataException {
    */
   public String deleteTimeseries(PartialPath pathPattern, boolean isPrefixMatch)
       throws MetadataException {
-    try {
-      List<MeasurementPath> allTimeseries = mtree.getMeasurementPaths(pathPattern, isPrefixMatch);
-      if (allTimeseries.isEmpty()) {
-        // In the cluster mode, the deletion of a timeseries will be forwarded to all the nodes. For
-        // nodes that do not have the metadata of the timeseries, the coordinator expects a
-        // PathNotExistException.
-        throw new PathNotExistException(pathPattern.getFullPath());
-      }
+    List<SGMManager> sgmManagers =
+        storageGroupManager.getInvolvedSGMManagers(pathPattern, isPrefixMatch);
+    if (sgmManagers.isEmpty()) {
+      // In the cluster mode, the deletion of a timeseries will be forwarded to all the nodes. For
+      // nodes that do not have the metadata of the timeseries, the coordinator expects a
+      // PathNotExistException.
+      throw new PathNotExistException(pathPattern.getFullPath());
```

Comment:
If a exception is threw here, these two scenarios are not distinguished:
 * This node actually doesn't have this series and it is expected.
 * The parameter is not correct for this method and it should return an exception

Reply:
This is the implementation of old cluster. It's just kept to help pass IT/UT of old cluster. This will be removed in future development of new cluster.